drivers/adc/adc_stm32.c

```diff
@@ -592,6 +594,7 @@ static int adc_stm32_calibrate(const struct device *dev)
 	defined(CONFIG_SOC_SERIES_STM32H7RSX) || \
 	defined(CONFIG_SOC_SERIES_STM32L0X) || \
 	defined(CONFIG_SOC_SERIES_STM32U0X) || \
+	defined(CONFIG_SOC_SERIES_STM32U3X) || \
```

Comment:
LL_ADC_REG_SetDMATransfer is not defined on U3. You should call LL_ADC_REG_SetDataTransferMode instead.